aks-node-controller/app.go

```diff
@@ -158,11 +163,9 @@ func (a *App) writeCompleteFileOnError(err error) {
 
 func (a *App) ProvisionWait(ctx context.Context, filepaths ProvisionStatusFiles) (string, error) {
 	if _, err := os.Stat(filepaths.ProvisionCompleteFile); err == nil {
-		data, err := os.ReadFile(filepaths.ProvisionJSONFile)
-		if err != nil {
-			return "", fmt.Errorf("failed to read provision.json: %w. One reason could be that AKSNodeConfig is not properly set", err)
-		}
-		return string(data), nil
+		// Fast path: provision.complete already exists when we enter. Avoid watcher overhead.
```

Comment:
A side note. It seems like there is a potential for a race.

What if the file created after we run `os.Stat` but before `NewWatcher`?

Reply:
Actually, solution is simpler:

Just move `os.Stat(filepaths.ProvisionCompleteFile)` after `watcher.Add`

Reply:
That's a good point. Updated and added unit tests.